sdk/search/search-documents/src/searchIndexingBufferedSender.ts

```diff
@@ -1,133 +1,438 @@
    * @param listener - Event Listener
    */
-  off(event: "beforeDocumentSent", listener: (e: IndexDocumentsAction<T>) => void): void;
+  public off(event: "beforeDocumentSent", listener: (e: IndexDocumentsAction<T>) => void): void;
   /**
    * Detach Batch Succeeded Event
    *
    * @param event - Event to be emitted
    * @param listener - Event Listener
    */
-  off(event: "batchSucceeded", listener: (e: IndexDocumentsResult) => void): void;
+  public off(event: "batchSucceeded", listener: (e: IndexDocumentsResult) => void): void;
   /**
    * Detach Batch Failed Event
    *
    * @param event - Event to be emitted
    * @param listener - Event Listener
    */
-  off(event: "batchFailed", listener: (e: RestError) => void): void;
+  public off(event: "batchFailed", listener: (e: RestError) => void): void;
+  /**
+   * Detach Batch Resizng Event
+   *
+   * @param event - Event to be emitted
+   * @param listener - Event Listener
+   */
+  public off(event: "batchResizing", listener: (e: RestError) => void): void;
+  public off(
+    event: "batchAdded" | "beforeDocumentSent" | "batchSucceeded" | "batchFailed" | "batchResizing",
+    listener: (e: any) => void
+  ): void {
+    this.emitter.removeListener(event, listener);
+  }
+
+  private isBatchReady(): boolean {
+    return this.batchObject.actions.length >= this.initialBatchActionCount;
+  }
+
+  private async internalFlush(force: boolean, options: OperationOptions = {}): Promise<void> {
+    if (force || (this.autoFlush && this.isBatchReady())) {
+      // Split it
+      const actions: IndexDocumentsAction<T>[] = this.batchObject.actions;
+      this.batchObject = new IndexDocumentsBatch<T>();
+      while (actions.length > 0) {
+        const actionsToSend = actions.splice(0, this.initialBatchActionCount);
+        await this.submitDocuments(actionsToSend, options);
+      }
+    }
+  }
+
+  private async submitDocuments(
+    actionsToSend: IndexDocumentsAction<T>[],
+    options: OperationOptions,
+    retryAttempt: number = 1
+  ): Promise<void> {
+    try {
+      for (const action of actionsToSend) {
+        this.emitter.emit("beforeDocumentSent", action);
+      }
+      const result = await this.client.indexDocuments(
+        new IndexDocumentsBatch<T>(actionsToSend),
+        options
+      );
+      // raise success event
+      this.emitter.emit("batchSucceeded", result);
+    } catch (e) {
+      if (e.code && e.code === "413") {
+        // Cut the payload size to half
+        let splitActionsArray = [];
+        splitActionsArray[0] = actionsToSend.slice(0, actionsToSend.length / 2);
+        splitActionsArray[1] = actionsToSend.slice(actionsToSend.length / 2, actionsToSend.length);
+        for (const actions of splitActionsArray) {
+          this.emitter.emit("batchResizing", {
+            oldBatchActionCount: actionsToSend.length,
+            newBatchActionCount: actions.length
+          });
+          this.submitDocuments(actions, options);
```

Comment:
This is the new code added to handle the payload too high (413) code. Kindly review it